ReSharper.FSharp/src/FSharp.Psi.Features/src/Daemon/Stages/FcsErrorsStageProcessBase.fs

```diff
@@ -80,9 +85,16 @@ type FcsErrorsStageProcessBase(fsFile, daemonProcess) =
     let createHighlighting (error: FSharpErrorInfo) (range: DocumentRange): IHighlighting =
         match error.ErrorNumber with
         | TypeEquation when error.Message.StartsWith(ifExprMissingElseBranch, StringComparison.Ordinal) ->
-            createHighlightingFromNodeWithMessage UnitTypeExpectedError range error 
-
-        | VarBoundTwice ->
+            createHighlightingFromNodeWithMessage UnitTypeExpectedError range error
+            
+        | NotAFunction ->
+            let notAFunctionNode = ExpressionSelectionUtil.GetExpressionInRange(fsFile, range, false, null) :> ISynExpr
```

Comment:
There're cases in this error to tree node mapping where we don't want to take an expression and want to take a tree node instead. I've added  a separate tree node selection provider for it to use here. If we are ever to change how we get these nodes, it'd be easier to change it in a single place, i.e. in the provider which encapsulates this logic.

```suggestion
            let notAFunctionNode = nodeSelectionProvider.GetExpressionInRange(fsFile, range, false, null) :> ISynExpr
```